src/main/java/org/radarbase/kafka/connect/transforms/KeyValueTransform.java

```diff
@@ -0,0 +1,125 @@
+/**
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.radarbase.kafka.connect.transforms;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.transforms.Transformation;
+import static org.apache.kafka.connect.transforms.util.Requirements.requireMap;
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+import java.util.*;
+
+/**
+ *
+ *
+ * This transforms records by copying the record key into the record value.
+ */
+public class KeyValueTransform<R extends ConnectRecord<R>> implements Transformation<R> {
+  private static final String PURPOSE = "copying fields from value to key";
+  private static final String TIMESTAMP_FIELD = "timestamp";
+  private static final String KEYVALUE_SCHEMA_NAME = "KeyToValue";
+  private static final String TIME_FIELD = "time";
+
+  @Override
+  public R apply(R r) {
+    if (r.valueSchema() == null) {
+      return applySchemaless(r);
+    } else {
+      return applyWithSchema(r);
+    }
+  }
+
+  private R applyWithSchema(R r) {
+    SchemaBuilder schemaBuilder = SchemaBuilder.struct().name(KEYVALUE_SCHEMA_NAME);
+    schemaBuilder = updateSchema(schemaBuilder, r.keySchema());
+    schemaBuilder = updateSchema(schemaBuilder, r.valueSchema());
+    schemaBuilder.field(TIMESTAMP_FIELD, Schema.INT64_SCHEMA);
+    Schema schema = schemaBuilder.build();
+
+    final Struct recordValue = requireStruct(r.value(), PURPOSE);
+    final Struct recordKey = requireStruct(r.key(), PURPOSE);
+    Struct newData = new Struct(schema);
+    newData = addValuesToNewSchema(newData, recordKey, r.keySchema());
+    newData = addValuesToNewSchema(newData, recordValue, r.valueSchema());
+    newData.put(TIMESTAMP_FIELD, r.timestamp());
+
+    return r.newRecord(r.topic(), r.kafkaPartition(), null, null, schema, newData, r.timestamp());
+  }
+
+  private Struct addValuesToNewSchema(Struct newData, Struct oldData, Schema oldSchema){
+    for (Field field: oldSchema.fields()) {
+      String fieldName = field.name();
+      Object fieldVal = oldData.get(field);
+      if(fieldName.contains(TIME_FIELD))
```

Comment:
To avoid parsing unrelated fields that have the substring `time`, perhaps explicitly check for time, timeReceived, timeCompleted and whether the time field is a double.

Reply:
Ok I've added all of time fields to a set.